src/Microsoft.DotNet.ProjectJsonMigration/MigrationBackupPlan.cs

```diff
@@ -12,27 +12,36 @@ namespace Microsoft.DotNet.ProjectJsonMigration
     internal class MigrationBackupPlan
     {
         private readonly FileInfo globalJson;
+        private readonly Dictionary<DirectoryInfo, IEnumerable<FileInfo>> mapOfProjectBackupDirectoryToFilesToMove;
```

Comment:
For an important reason that was revealed by existing tests. Consider this super-common project structure:

```
root
    |
    global.json
    |
    src
        |
        project1
        |   |
        |   project.json
        |
        project2
            |
            project.json
```

If we process the projects separately, we end up creating two backup folders due to the unique directory name logic:

```
root
    |
    backup
    |   |
    |   global.json
    |   |
    |   src
    |       |
    |       project1
    |           |
    |           project.json
    |
    backup_1
    |   |
    |   src
    |       |
    |       project2
    |           |
    |           project.json
    src
        |
        project1
        |   |
        |   project1.csproj
        |
        project2
            |
            project2.csproj
```

By processing everything at once, it's much easier to create the desired structure (while still keeping the unique backup folder name for situations where a backup folder already exists).

```
root
    |
    backup
    |   |
    |   global.json
    |   |
    |   src
    |       |
    |       project1
    |       |   |
    |       |   project.json
    |       |
    |       project2
    |           |
    |           project.json
    src
        |
        project1
        |   |
        |   project1.csproj
        |
        project2
            |
            project2.csproj
```